app/src/main/java/com/picone/go4lunch/presentation/ui/AuthenticationFragment.java

```diff
@@ -55,9 +55,9 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
         initAuthenticationAborted();
-        mUserViewModel.getCurrentUser().observe(getViewLifecycleOwner(), currentUser -> {
+        mUserViewModel.getCurrentUser.observe(getViewLifecycleOwner(), currentUser -> {
             if (isNewUser) mUserViewModel.addUser(currentUser);
-            mUserViewModel.getAddUserState().observe(getViewLifecycleOwner(), addUserState -> {
```

Comment:
Nested livedata subscription.

Reply:
the things i have seen on the net call of mediator live data or Transformations.switchMap method in view model. Is this what you're thinking about?

Reply:
Done